infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftEnvironmentFactory.java

```diff
@@ -146,6 +159,32 @@ protected OpenShiftEnvironment doCreate(
     return osEnv;
   }
 
+  @VisibleForTesting
+  void setRamLimitAttribute(Map<String, InternalMachineConfig> machines, Collection<Pod> pods) {
+    for (Pod pod : pods) {
+      for (Container container : pod.getSpec().getContainers()) {
+        final String machineName = Names.machineName(pod, container);
+        final InternalMachineConfig machineConfig = machines.get(machineName);
+        if (isNullOrEmpty(machineConfig.getAttributes().get(MEMORY_LIMIT_ATTRIBUTE))) {
+          long recipeRamLimit = 0;
+          final ResourceRequirements resources = container.getResources();
+          final Quantity quantity;
+          if (resources != null
+              && resources.getLimits() != null
+              && (quantity = resources.getLimits().get("memory")) != null
+              && quantity.getAmount() != null) {
+            recipeRamLimit = KubernetesSize.toBytes(quantity.getAmount());
+          }
+          if (recipeRamLimit > 0) {
```

Comment:
I'm not sure I follow what is happening here. @akorneta can you correct me if I'm mistaken? Looks like here we put an amount of RAM into machines attributes from the recipe if it is set in the recipe. But AFAIK priorities are opposite - machine's attribute overrides the value from the recipe.

Reply:
yes, you're right, but we will get into this section only if the [condition](https://github.com/eclipse/che/pull/8228/files/b9902984e101a349acf8553b032ff092f11e808f#diff-27f70b17b68377e07c7362feb76f9102R168) is true, and this, in turn, means that particular machine configuration does not contain ram attribute.